sklearn/decomposition/truncated_svd.py

```diff
@@ -50,6 +53,18 @@ class TruncatedSVD(BaseEstimator, TransformerMixin):
         (scipy.sparse.linalg.svds), or "randomized" for the randomized
         algorithm due to Halko (2009).
 
+    whiten : bool, default = False
+        When True (False by default) the transformed vectors are multiplied
+        by the square root of (n_samples) and divided by the singular values to
+        ensure uncorrelated outputs with unit component-wise variances.
+
+        Whitening will remove some information from the transformed signal
+        (the relative variance scales of the components) but can sometime
+        improve the predictive accuracy of the downstream estimators by
+        making their data respect some hard-wired assumptions.
```

Comment:
could you explain this a bit more? This docstring seems to suggest more that "you might want to try both".  I don't think I understand whitening well enough, but I did some googling and I want to ask:

- [this page](http://ufldl.stanford.edu/wiki/index.php/Whitening) suggests that is can be good for some learning algorithms to receive features that are uncorrelated *and* unit-variance. In our context, the user already is looking at TruncatedSVD, so the user gets the "uncorrelated" part either way, right? The `whiten` flag just enforces unit variance.

- "some hard-wired assumptions": there is nothing beyond uncorrelated and unit variance, right? This makes it seem like there may be more that we are not saying.

 - It would be great to list some examples of algorithms for which this matters, vs algorithms for which is makes no difference. I would speculate, like, SGD vs decision trees?

(and perhaps the most important point that I am struggling to understand:)

If a data scientist is using TruncatedSVD simply to visualize / interpret the structure of a large scale dataset, *does it ever makes sense to whiten*?

Thanks for helping me understand this better